### ISSUE TITLE
Values of Multiselects are not persisted if form validation fails

### ISSUE THREAD

**Issue by richard-kramer** [trailblazer/formular#50]
I'm creating Users that have multiple regions through a Trailblazer operation and Reform. Doing so, my selected regions wont be selected any more, if the validation fails and the form gets re-rendered

controller
```
# app/controllers/users_controller.rb
class UsersController < ApplicationController
  def new
    result = run User::Create::Present
    render_cell(cell_object: result['contract.default'])
  end

  def create
    result = run User::Create

    if result.failure?
      render_cell(cell: User::Cell::New, cell_object: result['contract.default'])
    end
  end
end
```

models:
```
# app/models/region.rb
# == Schema Information
#
# Table name: regions
#
#  id         :integer          not null, primary key
#  name       :string(255)
#  created_at :datetime
#  updated_at :datetime
#

class Region < ApplicationRecord
  ## Associations

  has_and_belongs_to_many :users
end




# app/models/user.rb
# == Schema Information
#
# Table name: users
#
#  id                     :integer          not null, primary key
#  name                   :string(255)
#

class User < ApplicationRecord
  has_and_belongs_to_many :regions, autosave: true, uniq: true
end
```

operation:
```
# app/concepts/user/operation/create.rb
class User::Create < Trailblazer::Operation
  class Present < Trailblazer::Operation
    step Model(User, :new)
    step Contract::Build(constant: User::Contract::Create)
  end

  step Nested(Present)
  step Contract::Validate()
  step Contract::Persist()
end
```

contract:
```
# app/concepts/user/contract/create.rb
module User::Contract
  class Create < Reform::Form
    collection :regions

    validation do
      required(:regions).filled
    end
  end
end
```

cell:
```
# app/concepts/user/cell/new.rb
module User::Cell
  class New < Trailblazer::Cell
    include Formular::RailsHelper

    #############
    # current state/working
    #############
    # def regions
    #   data_for_select Region.all, model.regions
    # end
    #
    # def data_for_select(obj, selected = [])
    #   obj.map do |o|
    #     opt = [o.name, o.id]
    #     opt.push({ selected: 'selected' }) if (selected.include?(o) || selected.map(&:to_i).include?(o.id.to_i))
    #     opt
    #   end
    # end

    #############
    # desired state/not working, selected values get rendered as normal, unselected options
    #############
    def regions
      data_for_select Region.all
    end

    def data_for_select(obj)
      obj.map { |o| [o.name, o.id] }
    end
  end
end
```

cell template
```
# app/concepts/user/view/new.rb
= trb_form_for model, users_path do |f|
  = f.select :regions, label: "regions", collection: regions, multiple: true
```

**Comment by fran-worley** [comment]
You've got this issue because the standard select helper for identifying selected options works with a super simple `option_value == selected_value`. In your case, your `option_value` is a `region.id` (an Integer) and because you're not coercing your `selected_value` it will be a string as passed in from html params.

Rather than pollute our select helper with lots of magic to mask examples like this, you'd be better off coercing your `regions` attribute to an array of integers, rather than strings. Then you should find that your selects pick up correctly.

```ruby
require 'reform/form/coercion'
# if you're going to use this more than once, stick it in an initialiser otherwise stick 
# use Types::Form::Array.of(Form::Nil | Form::Int) directly in your property call
module Disposable::Twin::Coercion::Types
  IntArray = Form::Array.of(Form::Nil | Form::Int) 
end

module User::Contract
  class Create < Reform::Form
    include Reform::Form::Coercion
    property :region_ids, type: Types::IntArray, default: [] # by all means revert back to `regions`
    
    validation do
      # if you find that you're getting empties you may need to compact your array
      required(:region_ids).filled(:array?).each(:int?) 
    end
  end
end
```

If you are still finding that selects aren't picking up your values when coerced give us a shout and I'll take another look.

**Comment by fran-worley** [comment]
@richard-kramer this should be fixed in 0.2.3 just released.
